### ISSUE TITLE
Run flow analysis in script files

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#3764]
We don't detect use before assignment.  For example,

```C#
{
    int unassigned;
    int error = unassigned;
}
```

Note that the variables are scoped and, therefore, don't need to be promoted to fields.

**Comment by amcasey** [comment]
Presently, the default value is used (i.e. it's not actually a problem at runtime).

**Comment by amcasey** [comment]
Appears to have been fixed.
